--- a/cpp/src/arrow/util/config.h.cmake	2026-06-02 18:45:55
+++ b/cpp/src/arrow/util/config.h.cmake	2026-06-02 18:46:01
@@ -27,7 +27,7 @@
 
 #define ARROW_CXX_COMPILER_ID "@CMAKE_CXX_COMPILER_ID@"
 #define ARROW_CXX_COMPILER_VERSION "@CMAKE_CXX_COMPILER_VERSION@"
-#define ARROW_CXX_COMPILER_FLAGS "@CMAKE_CXX_FLAGS@"
+#define ARROW_CXX_COMPILER_FLAGS ""
 
 #define ARROW_BUILD_TYPE "@UPPERCASE_BUILD_TYPE@"
 
