--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -268,11 +268,9 @@
 FOREACH (file ${HEADERS})
   get_filename_component(apath ${file} PATH)
   get_filename_component(aname ${file} NAME)
   file(RELATIVE_PATH relpath ${CMAKE_SOURCE_DIR}/src/core ${apath})
-  IF (NOT aname MATCHES "^_.*")
     install(FILES ${file}
         DESTINATION include/${relpath}
         COMPONENT development)
-  ENDIF (NOT aname MATCHES "^_.*")
 ENDFOREACH (file)
 
 #install clucene-shared headers.
